### PR TITLE
feat: add `vm_error` to analysis failure receipts

### LINKED ISSUE #3328: More information from analysis errors
When attempting to deploy a contract, if the analysis fails, there is no information provided about the error in the transaction result, just a result of `(err none)`. Ex. [Transaction in explorer](https://explorer.stacks.co/txid/0xce6e56d0392c64acdd3240b7135b04268492d2eff5c2b913d3ce272ff5056425?chain=testnet]), [API retrieving transaction](https://stacks-node-api.testnet.stacks.co/extended/v1/tx/0xce6e56d0392c64acdd3240b7135b04268492d2eff5c2b913d3ce272ff5056425)

It would be nice to get more information about the error.

This is related to #3154

### PR DESCRIPTION
**Pull request by obycode**
### Applicable issues
- fixes #3328

### Checklist
- [x] Test coverage for new or modified code paths
- [x] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Review by kantai — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,8 @@ and this project adheres to the versioning scheme outlined in the [README.md](RE
 
 - The transaction receipts for smart contract publish transactions now indicate
   a result of `(err none)` if the top-level code of the smart contract contained
-  runtime error. Fixes issue #3154.
+  runtime error and include details about the erro in the `vm_error` field of
```
Comment (kantai) [inline CHANGELOG.md:14]:
```suggestion
  runtime error and include details about the error in the `vm_error` field of
```

Comment (obycode) [reply CHANGELOG.md]: Oops! Fixed in 78b9a8c504ed97880d94c232f762608ee7e296d0.

File: src/chainstate/stacks/db/transactions.rs [hunk 2]
```diff
@@ -2148,6 +2173,90 @@ pub mod test {
         }
     }
 
+    #[test]
+    fn process_smart_contract_transaction_syntax_error() {
+        let contracts = [
+            "(define-data-var bar int 0)) ;; oops",
+            ";; `Int` instead of `int`
+             (define-data-var bar Int 0)",
```
Comment (pavitthrap) [inline src/chainstate/stacks/db/transactions.rs:2181]: suggestion to include tests for both analysis & parse errors

**Review by pavitthrap — APPROVED** [review]
lgtm

File: src/chainstate/stacks/db/transactions.rs (hunk 2 above)
Comment (obycode) [reply src/chainstate/stacks/db/transactions.rs:2181]: The first test here is a parse error and the second test is an analysis error.

**Pull request merged**